contracts/multisig-prover/src/execute.rs

```diff
@@ -283,11 +284,23 @@ pub fn confirm_worker_set(deps: DepsMut, sender: Addr) -> Result<Response, Contr
     CURRENT_WORKER_SET.save(deps.storage, &worker_set)?;
     NEXT_WORKER_SET.remove(deps.storage);
 
-    Ok(Response::new().add_message(wasm_execute(
+    let mut response = Response::new().add_message(wasm_execute(
         config.multisig,
-        &multisig::msg::ExecuteMsg::RegisterWorkerSet { worker_set },
+        &multisig::msg::ExecuteMsg::RegisterWorkerSet {
+            worker_set: worker_set.clone(),
+        },
         vec![],
-    )?))
+    )?);
+
+    response = response.add_message(wasm_execute(
+        config.monitoring,
+        &monitoring::msg::ExecuteMsg::RegisterActiveWorkerSet {
+            next_worker_set: worker_set,
+        },
+        vec![],
+    )?);
+
+    Ok(response)
```

Comment:
Can't you just do something like `Ok(Response::new().add_message(...).add_message(...)` ?